src/playermat/Playmat.ttslua

```diff
@@ -484,68 +488,102 @@ function unmarkSpawned(guid, force)
+    log('Not spawning tokens, relative coordinates are x: ' .. localpos.x .. ' z: ' .. localpos.z)
+  elseif not object.is_face_down and shouldSpawnTokens(object) then
     spawnTokensFor(object)
   end
 end
 
+---------------------------------------------------------
+-- investigator ID grabbing and stat tracker
+---------------------------------------------------------
+
 function maybeUpdateActiveInvestigator(card)
-  local cardMetadata = JSON.decode(card.getGMNotes()) or {}
-  if (cardMetadata.type == "Investigator") then
-    activeInvestigatorId = cardMetadata.id
-    updateStatToken(willpowerTokenGuid, cardMetadata.willpowerIcons)
-    updateStatToken(intellectTokenGuid, cardMetadata.intellectIcons)
-    updateStatToken(combatTokenGuid, cardMetadata.combatIcons)
-    updateStatToken(agilityTokenGuid, cardMetadata.agilityIcons)
+  local notes = JSON.decode(card.getGMNotes())
+  if notes ~= nil and notes.type == "Investigator" and notes.id ~= activeInvestigatorId then
+    activeInvestigatorId = notes.id
+    STAT_TRACKER.call("updateStats", {notes.willpowerIcons, notes.intellectIcons, notes.combatIcons, notes.agilityIcons})
+
+    -- change state of action tokens
+    local search = searchArea(self.positionToWorld({-1.1, 0.05, -0.27}), {4, 1, 1})
+    local small_token = nil
+    local state_table = {
+      ["Guardian"] = 1,
+      ["Seeker"]   = 2,
+      ["Rogue"]    = 3,
+      ["Mystic"]   = 4,
+      ["Survivor"] = 5,
+      ["Neutral"]  = 6
+    }
+    
+    for _, obj in ipairs(search) do
+      local obj = obj.hit_object
+      if obj.getDescription() == "Action Token" and obj.getStateId() > 0 then
+        if self.positionToLocal(obj.getPosition()).x > -0.95 then
+          small_token = obj
+        else
+          objState(obj, state_table[notes.class])
+        end
+      end
+    end
+
+    -- update the small token with special action for certain investigators
+    -- Ursula Downs: Investigate action
+    if activeInvestigatorId == "04002" then
+      objState(small_token, 8)
+    -- Daisy Walker (only for normal front, not parallel): Tome action
+    elseif activeInvestigatorId == "01002" or activeInvestigatorId == "01002-r" or activeInvestigatorId == "01002-pb" then
+      objState(small_token, 9)
+    -- Tony Morgan: Engage/Fight action
+    elseif activeInvestigatorId == "06003" then
+      objState(small_token, 10)
+    -- Finn Edwards: Evade action
+    elseif activeInvestigatorId == "04003" then
+      objState(small_token, 11)
+    -- Bob Jenkins: Play Item action
+    elseif activeInvestigatorId == "08016" then
+      objState(small_token, 14)
+    else
+      objState(small_token, state_table[notes.class])
+    end
   end
 end
 
-function updateStatToken(tokenGuid, val)
-  local statToken = getObjectFromGUID(tokenGuid)
-  if (statToken == nil) then
-    return
-  end
-  statToken.call("reset_val")
-  for i = 1, val do
-    statToken.call("add_subtract", { alt_click = false })
-  end
+function objState(obj, stateId)
```

Comment:
setActionTokenState (since that's what it's used for) or setObjectState at a minimum

Reply:
Will rename